src/Driver/SQLite3/Exception.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Doctrine\DBAL\Driver\SQLite3;
+
+use Doctrine\DBAL\Driver\AbstractException;
+use Exception as CoreException;
+
+/**
+ * @internal
+ *
+ * @psalm-immutable
+ */
+final class Exception extends AbstractException
+{
+    public static function new(CoreException $exception): self
+    {
+        return new self($exception->getMessage(), null, (int) $exception->getCode(), $exception);
```

Comment:
Isn't the exception code already an integer? ([documentation](https://www.php.net/manual/en/exception.getcode.php))

Reply:
Unfortunately, we are dealing with the generic `\Exception` class here and there are child classes of `Exception` where `getCode()` does not return an int. The PHP manual states:

> Returns the exception code as int in Exception but possibly as other type in Exception descendants (for example as string in PDOException).

We can rule out the case here because we know we're only dealing with exceptions coming from `ext-sqlite3`, however Psalm can't know that. So either I perform this cast here or I have to tell Psalm to ignore a (probably false-positive) error. You'll find other places in the DBAL codebase where we decided in favor of the cast.